python/pyiceberg/avro/lazydict.py

```diff
@@ -0,0 +1,56 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import (
+    Dict,
+    Iterator,
+    Mapping,
+    Tuple,
+)
+
+
+class LazyDictIntInt(Mapping[int, int]):
```

Comment:
Could we generalize this one outside of Avro? If so, we could also move it to `utils/typedef.py`

Reply:
Should we throw in some unit-tests for the lazydict?

Reply:
Added.